src/scheduler.cc

```diff
@@ -19,7 +19,15 @@ namespace detail {
 
 	void abstract_scheduler::schedule() {
 		graph_serializer serializer([this](command_pkg&& pkg) {
-			if(m_is_dry_run && pkg.get_command_type() != command_type::epoch) { return; }
+			if(m_is_dry_run && pkg.get_command_type() != command_type::epoch && pkg.get_command_type() != command_type::horizon
+			    && pkg.get_command_type() != command_type::fence) {
+				// in dry runs, skip everything except epochs, horizons and fences
+				return;
+			}
+			if(m_is_dry_run && pkg.get_command_type() == command_type::fence) {
+				CELERITY_WARN("Encountered a \"fence\" command while \"CELERITY_DRY_RUN_NODES\" is set. "
```

Comment:
Should we print this every time a fence is encountered or just once? In general I think we might want to have some "warn once" utility.

Reply:
I agree that a warn once utility would be nice, but I think it should probably be a separate PR.